libraries/lib-sample-track/Mix.h

```diff
@@ -271,5 +253,86 @@ class SAMPLE_TRACK_API Mixer {
    // Last step is accumulating results into the output buffers, with dithering
 };
 
+//! Fetches from tracks, applies envelopes; can resample, and warp time, even
+//! backwards, as for scrubbing.
+/*!
+ This class inherits AudioGraph::Source but does not yet fulfill the contracts
+ of all of the members.  But it is not yet used through any pointer or
+ reference to its base class.
+ */
+class MixerSource final : public AudioGraph::Source {
+public:
+   using TimesAndSpeed = MixerOptions::TimesAndSpeed;
+
+   /*!
+    @pre `pTimesAndSpeed != nullptr`
+    */
+   MixerSource(const SampleTrack &leader, size_t i,
```

Comment:
Generally I'm not a fan of such a long arguments lists,

Reply:
Of course not, and the purpose of later commits is to shorten them after that rough separation